### PR TITLE
8302518: Add missing Op_RoundDoubleMode in VectorNode::vector_operands()

### PR DESCRIPTION
**Pull request by pfustc**
This is a one-line fix.

`VectorNode::vector_operands()` is a C2 function used to get a half-open range of `[start, end)` defining which operands are vectors to process. `Op_RoundDoubleMode` has two operands but only the 1st one is a vector. Its 2nd operand is the rounding mode. Missing this op results in its 2nd operand treated as a vector. It is fortunate that it hasn't caused any issue yet as the code follows just treated the 2nd input as an immediate to replicate later. If we add some check like `n->is_Vector()` for the vectorized node then, there will be issues. We should fix this potential problem.

Tested tier1.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8302518](https://bugs.openjdk.org/browse/JDK-8302518): Add missing Op_RoundDoubleMode in VectorNode::vector_operands()

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [53038661](https://git.openjdk.org/jdk/pull/12584/files/5303866122f3f58cac2d6cc348a9c48f28cd8146)
 * [Jatin Bhateja](https://openjdk.org/census#jbhateja) (@jatin-bhateja - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12584/head:pull/12584` \
`$ git checkout pull/12584`

Update a local copy of the PR: \
`$ git checkout pull/12584` \
`$ git pull https://git.openjdk.org/jdk pull/12584/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12584`

View PR using the GUI difftool: \
`$ git pr show -t 12584`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12584.diff">https://git.openjdk.org/jdk/pull/12584.diff</a>

</details>

### CODE REVIEW

**Review by vnkozlov — APPROVED** [review]
Looks good to me.

**Comment by vnkozlov** [comment]
@jatin-bhateja, as author of this code, please look.

File: src/hotspot/share/opto/vectornode.cpp [hunk 1]
```diff
@@ -551,6 +551,7 @@ void VectorNode::vector_operands(Node* n, uint* start, uint* end) {
   case Op_LShiftI:  case Op_LShiftL:
   case Op_RShiftI:  case Op_RShiftL:
   case Op_URShiftI: case Op_URShiftL:
+  case Op_RoundDoubleMode:
```
Comment (jatin-bhateja) [inline src/hotspot/share/opto/vectornode.cpp:554]: Can you also add a check for Op_RotateRight and Op_RotateLeft here, its count argument (second operand) is treated as a vector/scalar based on following condition
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/superword.cpp#L2609.

Comment (pfustc) [reply src/hotspot/share/opto/vectornode.cpp:554]: Thanks for comments. I added those opcodes with a condition in latest commit.

**Pull request closed without merging**